sulat/scoring.py

```
@@ -0,0 +1,540 @@
+            except ValueError:
+                pass
+
+    elif isinstance(s, datetime.date):
+        return s
+    elif isinstance(s, datetime.datetime):
+        return s.date()
+    
+    return None
+
+
+def _score_text(g, p, field_name: str = None):
```

Ruff (0.13.1)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)